schema/schema_test.go

```diff
@@ -0,0 +1,295 @@
+	}
+	return columns
+}
+
+func buildColumnOperationBatch(num int, operation, inboundNameKey, outboundNameKey, inboundName, outboundName, transformer, columnCreationOptions string) ColumnOperation {
+	return ColumnOperation{
+		Operation:    operation,
+		InboundName:  fmt.Sprintf(inboundNameKey+"%d", num),
+		OutboundName: fmt.Sprintf(outboundNameKey+"%d", num),
+		NewColumnDefinition: ColumnDefinition{
+			InboundName:           fmt.Sprintf(inboundName+"%d", num),
+			OutboundName:          fmt.Sprintf(outboundName+"%d", num),
+			Transformer:           transformer,
+			ColumnCreationOptions: columnCreationOptions,
+		},
+	}
+}
+
+func buildColumnOperations(start, end int, operation, inboundNameKey, outboundNameKey, inboundName, outboundName, transformer, columnCreationOptions string) []ColumnOperation {
+	var columnOperations []ColumnOperation
+	for i := start; i < end; i++ {
+		columnOperations = append(columnOperations, buildColumnOperationBatch(i, operation, inboundNameKey, outboundNameKey, inboundName, outboundName, transformer, columnCreationOptions))
+	}
+	return columnOperations
+}
+
+func buildTableOption(distKey, sortKey []string) TableOption {
+	return TableOption{
+		DistKey: distKey,
+		SortKey: sortKey,
+	}
+}
+
+func buildStringList(list ...string) []string {
+	if list == nil {
+		return []string{}
+	}
+	return list
+}
+
+func buildColumnSchema(tableOption TableOption, columns []ColumnDefinition) ColumnSchema {
+	return ColumnSchema{
+		TableOption: tableOption,
+		Columns:     columns,
+	}
+}
+
+func buildMigration(tableOperation, name string, columnOperations []ColumnOperation, tableOption TableOption) Migration {
+	return Migration{
+		TableOperation:   tableOperation,
+		Name:             name,
+		ColumnOperations: columnOperations,
+		TableOption:      tableOption,
+	}
+}
+
+func TestAddTable(t *testing.T) {
+	t.Log("Testing Add Table")
+	t.Log("Initializing boilerplate")
+
+	emptyEvent := Event{
+		Name:    "empty",
+		Version: 1,
+	}
+
+	simpleColumnOperations := buildColumnOperations(0, 3, "add", "simpleColIn", "simpleColOut", "simpleColIn", "simpleColOut", "varchar", "(30)")
+	simpleTableOption := buildTableOption(buildStringList(simpleColumnOperations[0].NewColumnDefinition.OutboundName), buildStringList())
+	simpleMigration := buildMigration("add", "empty", simpleColumnOperations, simpleTableOption)
+
+	simpleColumns := buildColumns(0, 3, "simpleColIn", "simpleColOut", "varchar", "(30)")
+	simpleColumnSchema := buildColumnSchema(simpleTableOption, simpleColumns)
+	expectedEvent := buildEvent("empty", 2, simpleColumnSchema, simpleMigration)
+
+	t.Log("Starting Generic Test that should Migrate successfuly")
+	migrator := BuildMigratorBackend(simpleMigration, emptyEvent)
+	newEvent, err := migrator.ApplyMigration()
+	if err != nil {
+		t.Log(err.Error())
+		t.Log("err was incorrect, should have passed")
+		t.Fail()
```

Comment:
Above 3 lines can be simplified to:
t.Fatalf("%s\nerr was incorrect, should have passed", err.Error())

Might read better as: "Unexpected error: %s"